src/StreamFactory/DiactorosStreamFactory.php

```diff
@@ -24,7 +24,7 @@ public function createStream($body = null)
             } else {
                 $stream = new Stream('php://memory', 'rw');
 
-                if (null !== $body) {
+                if (null !== $body || '' !== $body) {
```

Comment:
@dbu indeed, an empty stream does not have to be rewinded, or does it?

@joelwurtz According to the interface the allowed types are `string|resource|StreamInterface|null`, so an object would not be valid in the first place, you would have to cast it to string first.